server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -2097,13 +2108,16 @@ void handleBonusPools(Owner owner, Map<String, PoolQuantity> pools,
         public void handlePostEntitlement(PoolManager manager, Consumer consumer,
             Map<String, Entitlement> entitlements) {
             Set<String> stackIds = new HashSet<String>();
+
             for (Entitlement entitlement : entitlements.values()) {
                 if (entitlement.getPool().isStacked()) {
                     stackIds.add(entitlement.getPool().getStackId());
                 }
             }
             List<Pool> subPoolsForStackIds = null;
-            if (!stackIds.isEmpty()) {
+            // Manifest and Share consumers should not contribute to the sharing org's stack,
+            // as Share consumers should not have created a stack derived pool in the first place.
```

Comment:
@vritant ^^ Not sure if you seen that from yesterday. Looks like it is the last thing to address?